lib/parsing/ast.hpp

```diff
@@ -88,22 +88,28 @@ namespace ast {
     };
 
     struct ExistsSentence {
-        std::vector<TypedList<Variable>> variables;
+        TypedList<Variable> variables;
         Sentence sentence;
     };
 
     struct ForallSentence {
-        std::vector<TypedList<Variable>> variables;
+        TypedList<Variable> variables;
         Sentence sentence;
     };
 
+    struct Action : x3::position_tagged {
+        Name name;
+        TypedList<Variable> parameters;
+        Sentence precondition;
+    };
+
     struct Domain : x3::position_tagged {
         Name name;
         std::vector<std::string> requirements;
         TypedList<Name> types;
         TypedList<Name> constants;
         std::vector<AtomicFormulaSkeleton> predicates;
-        // std::vector<Action> actions;
+        std::vector<Action> action;
```

Comment:
Can you change `action` to `actions` here to reflect that there are potentially multiple actions associated with a domain?

Reply:
Done